### PR TITLE
Allow users to create and use shielded spaces

### PR DESCRIPTION
**Pull request by sigmavirus24**
This allows users to create a shielded (HIPAA-compliant) Heroku space with this provider.

### CODE REVIEW

File: heroku/resource_heroku_space.go [hunk 1]
```diff
@@ -67,6 +73,7 @@ func resourceHerokuSpaceCreate(d *schema.ResourceData, meta interface{}) error {
 		vs := v.(string)
 		opts.Region = &vs
 	}
+	opts.Shield = d.Get("shield").(bool)
```
Comment (joestump) [inline heroku/resource_heroku_space.go]: This line appears to be breaking tests.

Comment (joestump) [reply heroku/resource_heroku_space.go]: This change fixes the build, but there's still code missing in the update/refresh methods as well.

```diff
-       opts.Shield = d.Get("shield").(bool)
+
+       if v := d.Get("shield"); v != nil {
+               vs := v.(bool)
+               log.Printf("[DEBUG] Shield enabled: %t", vs)
+               opts.Shield = &vs
+       }
```

Comment (sigmavirus24) [reply heroku/resource_heroku_space.go]: There's no way to `update` a space to use `shield`. Users need to recreate them. I'll investigate the test failures on monday

File: heroku/resource_heroku_space.go [hunk 2]
```diff
@@ -45,6 +45,12 @@ func resourceHerokuSpace() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"shield": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
```
[the remark this answers is not in the document]
Comment (sigmavirus24) [reply heroku/resource_heroku_space.go]: Ah, thanks for that tip!

**Comment by joestump** [comment]
@sheax0r we should target this one along with the exporting of NAT IPs in #46 for release after we get this current one out. I'll need to update the `heroku_space` resource as well to pull in NAT IPs.

**Comment by sheax0r** [comment]
@sigmavirus24 Sorry for the delay on reviewing this. I'm trying to make sure we get the broken spaces tests fixed before merging anym new features; we can't release until our tests pass anyway. 

Working on fixing broken tests with [this](https://github.com/terraform-providers/terraform-provider-heroku/pull/53) - hopefully we can nail this down soon and start releasing the plugin again.

**Comment by sigmavirus24** [comment]
I wasn't aware there were broken tests. I can maybe look at those sometime this week.

**Comment by sheax0r** [comment]
@sigmavirus24 Here's where I'm attempting to fix the broken tests so far. This PR fixes one of them, but changes the default behaviour of the shielded resource, defaulting trusted_ip_ranges to `[]`: https://github.com/terraform-providers/terraform-provider-heroku/pull/53

**Comment by joestump** [comment]
@sheax0r @sigmavirus24 #61 fixes acceptance tests. Would be nice to get this out. We need to add docs as well.

**Comment by joestump** [comment]
#63 added this functionality. Closing this out.

**Pull request closed without merging**